app/src/main/java/com/arctouch/codechallenge/home/data/repository/GenresRepository.kt

```diff
@@ -0,0 +1,30 @@
+package com.arctouch.codechallenge.home.data.repository
+
+import com.arctouch.codechallenge.core.api.TmdbApi
+import com.arctouch.codechallenge.core.domain.model.Genre
+import com.arctouch.codechallenge.home.data.cache.GenresCache
+
+class GenresRepository(private val api: TmdbApi) {
+
+    suspend fun getGenres(): List<Genre> {
+        return GenresCache.genres ?: getGenresFromApi()
+    }
+
+    private suspend fun getGenresFromApi(): List<Genre> {
+        return try {
+            val result = api.genresAsync().await()
+            val resultBody = result.body()
+            if (result.isSuccessful && resultBody != null) {
+                GenresCache.cacheGenres(resultBody.genres)
+                resultBody.genres
+            } else {
+                //TODO handle api error
```

Comment:
In a general way i dont think it´s a problem, can be very useful to describe things or as a reminder, it helps when working with teams, also if you use android studio integrated VCS its going to show an warning dialog before you commit changes

Reply:
For this case, is there because i didn´t have time to finish all that i planned to do, i had only two days to create the app, so my priority was to finish the main requisites, after that the optionals and later do general improvements/extras